evcache-core/src/main/java/com/netflix/evcache/pool/EVCacheClient.java

```diff
@@ -187,39 +188,36 @@ public Integer getMaxHashLength() {
         return this.maxHashLength.get();
     }
 
-    private Collection<String> validateReadQueueSize(Collection<String> canonicalKeys, EVCache.Call call) {
-        if (evcacheMemcachedClient.getNodeLocator() == null) return canonicalKeys;
-        final Collection<String> retKeys = new ArrayList<>(canonicalKeys.size());
-        for (String key : canonicalKeys) {
-            final MemcachedNode node = evcacheMemcachedClient.getNodeLocator().getPrimary(key);
-            if (node instanceof EVCacheNode) {
-                final EVCacheNode evcNode = (EVCacheNode) node;
-                if (!evcNode.isAvailable(call)) {
-                    continue;
-                }
+    private boolean validateReadQueueSize(MemcachedNode node, String key, EVCache.Call call) {
```

Comment:
nit:key can be removed? and from all the callers too.

Passing key here doesn't seem to serve any purpose. We want to iterate over all the keys and that's achieved when calling this.

Reply:
Agreed, fixed in c0c33cf !